xarray/core/coordinates.py

```diff
@@ -523,12 +533,16 @@ def _merge_inplace(self, other):
                 if k not in self.xindexes
             }
             variables, indexes = merge_coordinates_without_align(
-                [self, other], prioritized
+                [self, other], prioritized, compat=compat
             )
             yield
             self._update_coords(variables, indexes)
 
-    def merge(self, other: Mapping[Any, Any] | None) -> Dataset:
+    def merge(
+        self,
+        other: Mapping[Any, Any] | None,
+        compat: CompatOptions | CombineKwargDefault = "minimal",
```

Comment:
let's put these behind kwargs so no one thinks we can pass multiple things to merge?

```suggestion
        *,
        compat: CompatOptions | CombineKwargDefault = "minimal",
```